### PR TITLE
Troubleshooting and warning in code related to issues with native ZipArchiver

### LINKED ISSUE #2565: Failure if unzip is not installed (as happens on a clean Ubuntu 13.10 install)
## Short situation:

I was trying to do a create-project using composer on a Ubuntu 13.10 clean vm. PHP 5.5 installed using Chef with https://github.com/Kunstmaan/chef-php/blob/master/recipes/package.rb (irrelevant for the issue) that mostly just does apt-get install php5-cgi php5 php5-dev php5-cli php-pear, ext-zip is built in.

## The issue:

```
  - Installing doctrine/common (2.3.0)
Downloading https://api.github.com/repos/doctrine/common/zipball/d1c7d4334e38cad603a5c863d4c7b91bb04ec6b2
    Downloading: 100%         
Writing /home/vagrant/.composer/cache/files/doctrine/common/d1c7d4334e38cad603a5c863d4c7b91bb04ec6b2.zip into cache
    Extracting archive
Executing command (CWD): unzip '/home/projects/test/data/test/vendor/doctrine/common/a259fc31d4f248f87439a5515c6aa490' -d 'vendor/composer/25117f89' && chmod -R u+w 'vendor/composer/25117f89'
Reading ./composer.json


                                                             
  [RuntimeException]                                         
  There was an error extracting the ZIP file. Corrupt file?  
                                                             


Exception trace:
 () at phar://composer.phar/bin/../src/../src/Composer/Downloader/ZipDownloader.php:76
 Composer\Downloader\ZipDownloader->extract() at phar://composer.phar/bin/../src/../src/Composer/Downloader/ArchiveDownloader.php:43
 Composer\Downloader\ArchiveDownloader->download() at phar://composer.phar/bin/../src/../src/Composer/Downloader/DownloadManager.php:182
 Composer\Downloader\DownloadManager->download() at phar://composer.phar/bin/../src/../src/Composer/Installer/LibraryInstaller.php:155
 Composer\Installer\LibraryInstaller->installCode() at phar://composer.phar/bin/../src/../src/Composer/Installer/LibraryInstaller.php:86
 Composer\Installer\LibraryInstaller->install() at phar://composer.phar/bin/../src/../src/Composer/Installer/InstallationManager.php:153
 Composer\Installer\InstallationManager->install() at phar://composer.phar/bin/../src/../src/Composer/Installer/InstallationManager.php:140
 Composer\Installer\InstallationManager->execute() at phar://composer.phar/bin/../src/../src/Composer/Installer.php:510
 Composer\Installer->doInstall() at phar://composer.phar/bin/../src/../src/Composer/Installer.php:210
 Composer\Installer->run() at phar://composer.phar/bin/../src/../src/Composer/Command/CreateProjectCommand.php:176
 Composer\Command\CreateProjectCommand->installProject() at phar://composer.phar/bin/../src/../src/Composer/Command/CreateProjectCommand.php:143
 Composer\Command\CreateProjectCommand->execute() at phar://composer.phar/bin/../src/../vendor/symfony/console/Symfony/Component/Console/Command/Command.php:241
 Symfony\Component\Console\Command\Command->run() at phar://composer.phar/bin/../src/../vendor/symfony/console/Symfony/Component/Console/Application.php:881
 Symfony\Component\Console\Application->doRunCommand() at phar://composer.phar/bin/../src/../vendor/symfony/console/Symfony/Component/Console/Application.php:191
 Symfony\Component\Console\Application->doRun() at phar://composer.phar/bin/../src/../src/Composer/Console/Application.php:117
 Composer\Console\Application->doRun() at phar://composer.phar/bin/../src/../vendor/symfony/console/Symfony/Component/Console/Application.php:121
 Symfony\Component\Console\Application->run() at phar://composer.phar/bin/../src/../src/Composer/Console/Application.php:83
 Composer\Console\Application->run() at phar://composer.phar/bin/composer:43
 include() at /usr/local/bin/composer.phar:15
```

No clear indication what's wrong, but when trying to run the command by hand I noticed that "unzip" is not installed by default.

## The culprit in composer

https://github.com/composer/composer/blob/20a18dcb6068e3a05cf51927bd609f62539c4476/src/Composer/Downloader/ZipDownloader.php#L41

## The solution

For me to fix my problem, I just added "unzip" to my chef recipe for composer and everything works great. 

In general, I believe a test for the availability of the "unzip" command should be added

### PR DESCRIPTION
**Pull request by curry684**
Replaces #4964, fixes #2565 

(new PR since git crashed my other branch and I had to rebuild from master)

### CODE REVIEW

File: src/Composer/Downloader/ZipDownloader.php [hunk 1]
```diff
@@ -98,7 +98,10 @@ protected function extract($file, $path)
         }
 
         if (true !== $zipArchive->extractTo($path)) {
-            throw new \RuntimeException("There was an error extracting the ZIP file. Corrupt file?");
+            $this->io->writeError("<warn>As there is no 'unzip' command installed zip files are being unpacked using the PHP zip extension.</warn>");
+            $this->io->writeError("<warn> This may cause invalid reports of corrupted archives. Installing 'unzip' may remediate them.</warn>");
```
Comment (alcohol) [inline src/Composer/Downloader/ZipDownloader.php]: Is the extra indentation intentional here? Just checking :)

Comment (curry684) [inline src/Composer/Downloader/ZipDownloader.php]: Lol, nope, that's a c/p error from having to rebuild the PR hehe.

**Comment by alcohol** [comment]
Cheers

File: doc/articles/troubleshooting.md [hunk 2]
```diff
@@ -345,3 +345,14 @@ composer update
 ```
 
 See also https://github.com/composer/composer/issues/4180 for more information.
+
+## Zip archives are being reported as corrupted or not unpacked correctly.
+
+Composer can unpack zipballs using either a system-provided `unzip` utility or PHP's
+native `ZipArchiver` class, preferring the first. The `ZipArchiver` class however is
```
Comment (stof) [inline doc/articles/troubleshooting.md]: typo. the class is ``ZipArchive``, not ``ZipArchiver``

Comment (Seldaek) [inline doc/articles/troubleshooting.md]: Fixed

**Pull request merged**